.github/workflows/main-test.yml

```diff
@@ -0,0 +1,32 @@
+name: Test
+
+on:
+  pull_request:
+    branches:
+      - main
+
+env:
+  CARGO_TERM_COLOR: always
+  ARCH_TYPE: sm_70
+
+jobs:
+  test-golang-linux:
+    name: Test Golang on Linux
+    runs-on: [self-hosted, Linux, X64, icicle]
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+    - name: Build CUDA libs
+      run: |
+        git clone https://github.com/username/another-repo.git
+        cd another-repo/goicicle
```

Comment:
I'm still not sure what the purpose of this clone here is. It seems like we are trying to clone another repo, though it is a fake/non-existent repo?